rdbms/src/main/java/org/spine3/server/storage/jdbc/JdbcAggregateStorage.java

```diff
@@ -0,0 +1,441 @@
+
+        @SuppressWarnings("DuplicateStringLiteralInspection")
+        private static final String QUERY =
+                "CREATE TABLE IF NOT EXISTS %s (" +
+                    ID_COL + " %s, " +
+                    AGGREGATE_COL + " BLOB, " +
+                    SECONDS_COL + " BIGINT, " +
+                    NANOS_COL + " INT " +
+                ");";
+
+
+        protected CreateMainTableIfDoesNotExistQuery() {
+            super(QUERY);
+        }
+    }
+
+    private class CreateEventCountTableIfDoesNotExistQuery extends CreateTableIfDoesNotExistQuery {
+
+        @SuppressWarnings("DuplicateStringLiteralInspection")
+        private static final String QUERY =
+                "CREATE TABLE IF NOT EXISTS %s (" +
+                    ID_COL + " %s, " +
+                    EVENT_COUNT_COL + " BIGINT " +
+                ");";
+
+        protected CreateEventCountTableIfDoesNotExistQuery() {
+            super(QUERY);
+        }
+    }
+
+    private abstract class WriteEventCountQuery extends WriteQuery {
+
+        private final String query;
+        private final Id id;
+        private final int count;
+
+        protected WriteEventCountQuery(String query, Id id, int count) {
+            super(dataSource);
+            this.query = query;
+            this.id = id;
+            this.count = count;
+        }
+
+        @Override
+        protected PreparedStatement prepareStatement(ConnectionWrapper connection) {
+            final PreparedStatement statement = connection.prepareStatement(query);
+            try {
+                idColumn.setId(getIdIndexInQuery(), id, statement);
+                statement.setLong(getEventCountIndexInQuery(), count);
+                return statement;
+            } catch (SQLException e) {
+                throw new DatabaseException(e);
+            }
+        }
+
+        protected abstract int getIdIndexInQuery();
+
+        protected abstract int getEventCountIndexInQuery();
+
+        @Override
+        protected void logError(SQLException e) {
+            log().error("Failed to write the count of events after the last snapshot.", e);
+        }
+    }
+
+    private class InsertEventCountQuery extends WriteEventCountQuery {
+
+        @SuppressWarnings("DuplicateStringLiteralInspection")
+        private static final String INSERT_QUERY =
+                "INSERT INTO %s " +
+                " (" + ID_COL + ", " + EVENT_COUNT_COL + ')' +
+                " VALUES (?, ?);";
+
+        private InsertEventCountQuery(int count, Id id) {
+            super(format(INSERT_QUERY, eventCountTableName), id, count);
+        }
+
+        @Override
+        protected int getIdIndexInQuery() {
+            return 1;
+        }
+
+        @Override
+        protected int getEventCountIndexInQuery() {
+            return 2;
+        }
+    }
+
+    private class UpdateEventCountQuery extends WriteEventCountQuery {
```

Comment:
One thing that bugs me is that these queries are not adjustable to the Spine framework users.
I.e. what if I need to have a different table layout in my database? Or create indexes upon table creation to make the execution faster?

With the current approach the queries seem to be fixed and non-overridable.